### PR TITLE
precommit for firestore + profile

### PR DESCRIPTION
**Pull request by Destwood**

### CODE REVIEW

File: src/pages/Home/Home.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import { useTheme } from 'shared/theme';
+
+export const HomePage: React.FC = () => {
+	const auth = getAuth();
+	const { currentTheme, toggleTheme } = useTheme();
+
+	return (
+		<div>
+			<h1>
+				Home Page, current theme: {currentTheme} {currentTheme === 'light' ? 'асуждаю' : ''}
```
Comment (MAKSsMaK) [inline src/pages/Home/Home.tsx:14]: асуждаю

File: src/pages/Profile/style.css
Comment (Juliet-cyborg) [inline src/pages/Profile/style.css:1]: What is this file? For what?

File: src/services/user.services.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { doc, getDoc, setDoc } from 'firebase/firestore';
+import { firestore } from 'services/firebase.services';
+
+export class UserService {
+	static async createUserProfile(uid: string, userInfo: Record<string, any>) {
+		const userRef = doc(firestore, 'users', uid);
```
Comment (Juliet-cyborg) [inline src/services/user.services.ts]: I think this name needs to be made a constant in a separate file.

File: src/services/user.services.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+		await setDoc(userRef, userInfo);
+	}
+
+	static async getUserProfile(uid: string): Promise<Record<string, any> | null> {
+		const userRef = doc(firestore, 'users', uid);
```
Comment (Juliet-cyborg) [inline src/services/user.services.ts]: the same

File: src/shared/components/Button/ButtonStyle.tsx
Comment (Juliet-cyborg) [inline src/shared/components/Button/ButtonStyle.tsx:1]: if it's main styles for  buttons I think it must be in theme.ts (maybe u have other name for this file) file where is creation mui theme file

File: src/shared/modules/Header/Header.tsx [hunk 4]
```diff
@@ -48,11 +40,22 @@ const Header = () => {
...
 	// modal
-	const handleSubmit = async ({ email, password }: IAuth) => {
+	const handleSubmit = async ({ username, email, password }: IAuth) => {
 		try {
 			const userCredentials = isLogin ? await Auth.login(email, password) : await Auth.signUp(email, password);
+			const token = await userCredentials.user.getIdToken();
 
 			const { uid, email: username } = userCredentials.user;
+
+			localStorage.setItem('authToken', token);
```
Comment (Juliet-cyborg) [inline src/shared/modules/Header/Header.tsx]: put it into a constant

File: src/utils/constants/User.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+export const User = {
+	initInfo: {
+		id: '',
+		nickname: '',
+		email: '',
+		firstName: '',
+		lastName: '',
+		profilePicture: null,
+		games: [
```
Comment (Juliet-cyborg) [inline src/utils/constants/User.ts:13]: Do you need this mock data?

File: src/App.tsx [hunk 6]
```diff
@@ -1,27 +1,46 @@
...
-			<button onClick={toggleTheme}>Toggle Theme</button>
-		</div>
-	);
-};
+	useEffect(() => {
+		const fetchUserProfile = async () => {
+			Auth.restoreSession(async (userData) => {
+				if (userData) {
+					const { token, username, email, uid } = userData;
+					localStorage.setItem('authToken', token);
```
Comment (Juliet-cyborg) [inline src/App.tsx]: put it into a constant

File: src/App.tsx [hunk 7]
```diff
@@ -1,27 +1,46 @@
...
+					try {
+						const profile = await UserService.getUserProfile(uid);
+						console.log(profile);
+					} catch (error) {
+						console.error(`Failed to fetch user profile for UID ${uid}:`, error);
+					}
+
+					dispatch(setUser({ token, username, email }));
+				} else {
+					localStorage.removeItem('authToken');
```
Comment (Juliet-cyborg) [inline src/App.tsx]: the same

File: src/pages/Home/index.module.scss [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+
+.viewProfile {
+  color: #3b82f6;
+  font-size: 0.9rem;
+  margin-top: 5px;
+}
+
+.balance {
+  margin-top: 20px;
+  background-color: #374151;
```
Comment (Juliet-cyborg) [inline src/pages/Home/index.module.scss]: why do your haven`t colors in theme file and u just no import for it?

File: src/pages/Profile/style.css
Comment (Destwood) [reply src/pages/Profile/style.css:1]: tak treba 🤌

Comment (MAKSsMaK) [reply src/pages/Profile/style.css:1]: v naturi sho tze chomy css 🤏

File: src/shared/modules/Header/Header.tsx [hunk 9]
```diff
@@ -76,7 +91,7 @@ const Header = () => {
 			</div>
 			<div className="">
 				<div className={style.authButtons}>
-					{userFromStore.token ? (
+					{userData.id !== '' ? (
```
Comment (MAKSsMaK) [inline src/shared/modules/Header/Header.tsx]: why don't you just write userData.id ? (content) : (content), or check length userData.id.length

File: src/shared/modules/Header/Header.tsx [hunk 10]
```diff
@@ -48,20 +42,41 @@ const Header = () => {
...
-			dispatch(setUser({ token: uid, username }));
 			dispatch(togglePopup());
 		} catch (error) {
 			console.log(error);
 		}
 	};
 
 	const handleLogout = () => {
-		dispatch(clearUser());
+		console.log('logging out', userData);
```
Comment (MAKSsMaK) [inline src/shared/modules/Header/Header.tsx]: delete console

File: src/utils/constants/User.ts [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+export const User = {
+	firebaseKey: 'users',
+	initInfo: {
+		id: '',
+		username: '',
+		email: '',
+		firstName: '',
+		lastName: '',
+		profilePicture: null,
+		games: [
```
Comment (MAKSsMaK) [inline src/utils/constants/User.ts:13]: delete mock data if it is no longer needed

File: src/utils/constants/Auth.ts [hunk 12]
```diff
@@ -0,0 +1,17 @@
+import { IAuth } from 'types';
+
+const Auth = {
```
Comment (MAKSsMaK) [inline src/utils/constants/Auth.ts]: const Auth: IAuth = {object}, and delete as IAuth

File: src/utils/constants/User.ts [hunk 13]
```diff
@@ -0,0 +1,30 @@
+import { IUser } from 'types';
+import { Language, Themes } from 'utils/enums';
+
+export const User = {
```
Comment (MAKSsMaK) [inline src/utils/constants/User.ts]: The same const User: IUser = {}

File: src/store/User/Slice.ts [hunk 14]
```diff
@@ -1,10 +1,28 @@
...
 const initialState: IUser = {
-	email: '',
-	token: '',
 	id: '',
+	username: '',
+	email: '',
+	firstName: '',
+	lastName: '',
+	profilePicture: null,
+	games: [
```
Comment (MAKSsMaK) [inline src/store/User/Slice.ts:12]: delete mock data if it no longer needed

File: src/shared/theme/components/Switch.ts [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+	'& .MuiSwitch-switchBase.Mui-checked': {
+		transform: 'translateX(calc(100% - 0.75em ))',
+		'& + .MuiSwitch-track': {
+			backgroundPosition: '100%',
+		},
+		'& .MuiSwitch-thumb': {
+			backgroundPosition: '0',
+		},
+	},
+});
```
Comment (MAKSsMaK) [inline src/shared/theme/components/Switch.ts:40]: why didn't you move this to the main stylesheet for mui

File: src/shared/modules/Header/Header.tsx [hunk 16]
```diff
@@ -48,20 +42,41 @@ const Header = () => {
...
+
+		signOut(auth)
+			.then(() => {
+				console.log('signOutSuccsessful');
+				dispatch(clearUser());
+			})
+			.catch((e) => console.error(e));
+
+		localStorage.removeItem(tokenKey);
 	};
```
Comment (MAKSsMaK) [inline src/shared/modules/Header/Header.tsx:78]: const handleLogout = async () => {
  try {
    console.log('Logging out:', userData);
    await signOut(auth);
    console.log('Sign out successful');
    dispatch(clearUser());
    localStorage.removeItem(tokenKey);
    } catch (error) {
    console.error('Error during logout:', error);
  }
}; 
use await

File: src/shared/modules/Header/Header.tsx [hunk 17]
```diff
@@ -48,20 +42,41 @@ const Header = () => {
...
 
 	// modal
-	const handleSubmit = async ({ email, password }: IAuth) => {
+	const handleSubmit = async ({ username, email, password }: IAuth) => {
 		try {
 			const userCredentials = isLogin ? await Auth.login(email, password) : await Auth.signUp(email, password);
+			const token = await userCredentials.user.getIdToken();
+
+			const { uid } = userCredentials.user;
```
Comment (MAKSsMaK) [inline src/shared/modules/Header/Header.tsx]: const { user } = userCredentials;
    const token = await user.getIdToken();
    const { uid } = user;

File: src/pages/Profile/style.css [hunk 18]
```diff
@@ -0,0 +1,49 @@
+/* From Uiverse.io by cuzpq */
```
Comment (MAKSsMaK) [inline src/pages/Profile/style.css]: delete this file

File: src/App.tsx [hunk 19]
```diff
@@ -32,6 +51,7 @@ const App = () => {
 						<Sidebar />
 						<Routes>
 							<Route path="/" element={<HomePage />} />
+							<Route path="/profile" element={userData.id !== '' ? <Profile /> : <Navigate to="/" />} />
```
Comment (MAKSsMaK) [inline src/App.tsx]: userData.id ? :

File: src/pages/Home/Home.tsx [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
+
+export const HomePage: React.FC = () => {
+	//TODO - all styles and this page is temporary
+	const auth = getAuth();
+	const { currentTheme, toggleTheme } = useTheme();
+
+	return (
+		<div>
+			<h1>
+				Home Page, current theme: {currentTheme} {currentTheme === 'light' ? 'асуждаю' : ''}
```
Comment (MAKSsMaK) [inline src/pages/Home/Home.tsx:14]: асуждаю

File: src/pages/Home/index.module.scss [hunk 21]
```diff
@@ -0,0 +1,180 @@
+//TODO - all styles and this page is temporary
+
+.profilePage {
+  display: flex;
+  color: #f3f4f6;
```
Comment (MAKSsMaK) [inline src/pages/Home/index.module.scss]: use variables for all color

File: src/pages/Profile/Tabs/Security/Security.tsx [hunk 22]
```diff
@@ -0,0 +1,52 @@
...
+	const userData = useAppSelector(selectUser);
+	const auth = getAuth();
+	const user = auth.currentUser;
+	const userRef = doc(firestore, 'users', userData.id);
+
+	const handleDeleteAccount = async () => {
+		if (user) {
+			try {
+				await deleteDoc(userRef);
+				deleteUser(user)
```
Comment (MAKSsMaK) [inline src/pages/Profile/Tabs/Security/Security.tsx]: prosto await deleteUser(user) and thats all

File: src/pages/Profile/Tabs/UserSettings/UserSettings.tsx [hunk 23]
```diff
@@ -0,0 +1,71 @@
...
+
+						<div className={style.inputGroup}>
+							<label htmlFor="email">Email</label>
+							<Field id="email" name="email" type="email" placeholder="Enter your email" className={style.input} />
+							{touched.email && errors.email && <div className={style.error}>{errors.email}</div>}
+						</div>
+
+						<button type="submit" className={style.submitButton}>
+							Save Changes
+						</button>
```
Comment (MAKSsMaK) [inline src/pages/Profile/Tabs/UserSettings/UserSettings.tsx]: use nui button

File: src/utils/constants/User.ts (hunk 13 above)
Comment (Destwood) [reply src/utils/constants/User.ts:4]: constant User have string - firebaseKey and obj - initInfo
initInfo - using IUser, not User

File: src/utils/constants/User.ts (hunk 11 above)
Comment (Destwood) [reply src/utils/constants/User.ts:13]: Header, line 56, i use it as default info about user
if i need to delete it - should i try Redux or leave it like that?

File: src/pages/Profile/Tabs/UserSettings/UserSettings.tsx [hunk 24]
```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { getAuth } from 'firebase/auth';
+import { doc, updateDoc } from 'firebase/firestore';
+import { Field, Form, Formik } from 'formik';
+import { firestore } from 'services/firebase.services';
+import { Button } from 'shared/components';
+import { userScheme } from 'utils/schemas/profile';
+import style from './index.module.scss';
+
+interface UserSettingsValues {
```
Comment (MAKSsMaK) [inline src/pages/Profile/Tabs/UserSettings/UserSettings.tsx]: move interface in user interface file

File: src/pages/Profile/Tabs/UserSettings/UserSettings.tsx [hunk 25]
```diff
@@ -0,0 +1,70 @@
...
+	username: '',
+	lastName: '',
+	email: '',
+};
+
+export const UserSettings: React.FC = () => {
+	const auth = getAuth();
+
+	const handleSubmit = async (values: UserSettingsValues) => {
+		try {
```
Comment (MAKSsMaK) [inline src/pages/Profile/Tabs/UserSettings/UserSettings.tsx]: move this logic on services

**Pull request merged**